packages/records/lib/models/records.ts

```diff
@@ -471,9 +471,7 @@ export async function markNonCurrentGenerationRecordsAsDeleted({
                 sync_id: syncId,
                 deleted_at: null
             })
-            .whereNot({
-                sync_job_id: generation
-            })
```

Comment:
note it can have an impact, I don't know if PG can use the index the same way. Did you check with a dummy query in prod?